app/repositories/v2/ListRepository.scala

```diff
@@ -0,0 +1,141 @@
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import com.google.inject.Inject
+import com.mongodb.client.model.InsertManyOptions
+import config.AppConfig
+import models.FilterParams
+import models.GenericListItem
+import models.ListName
+import models.VersionId
+import org.mongodb.scala.bson.BsonValue
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.Indexes.ascending
+import org.mongodb.scala.model.Indexes.compoundIndex
+import org.mongodb.scala.model._
+import play.api.libs.json.JsObject
+import repositories.FailedWrite
+import repositories.ListRepositoryWriteResult
+import repositories.SuccessfulWrite
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.Codecs
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
+
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+@Singleton
+class ListRepository @Inject() (
+  mongoComponent: MongoComponent,
+  config: AppConfig
+)(implicit ec: ExecutionContext)
+    extends PlayMongoRepository[GenericListItem](
+      mongoComponent = mongoComponent,
+      collectionName = "v2-reference-data-lists-new",
+      domainFormat = GenericListItem.format,
+      indexes = ListRepository.indexes(config),
+      replaceIndexes = config.replaceIndexes
+    ) {
+
+  override lazy val requiresTtlIndex: Boolean = config.isTtlEnabled
+
+  def getListByName(listName: ListName, versionId: VersionId): Source[JsObject, NotUsed] = {
+    val filter = Aggregates.filter(
+      Filters.and(
+        Filters.eq("listName", listName.listName),
+        Filters.eq("versionId", versionId.versionId)
+      )
+    )
+
+    val projection = Aggregates.project(
+      Projections.fields(
+        Projections.include("data"),
+        Projections.exclude("_id")
+      )
+    )
+
+    Source.fromPublisher(
+      collection
+        .aggregate[BsonValue](Seq(filter, projection))
+        .allowDiskUse(true)
+        .map(Codecs.fromBson[JsObject](_))
+    )
+  }
+
+  def getListByNameWithFilter(listName: ListName, versionId: VersionId, filter: FilterParams): Source[JsObject, NotUsed] = {
+
+    val standardFilters: Bson = Aggregates.filter(
+      Filters.and(
+        Filters.eq(fieldName = "listName", value = listName.listName),
+        Filters.eq(fieldName = "versionId", value = versionId.versionId)
+      )
+    )
+
+    val extraFilters: Seq[Bson] = filter.parameters.map(
+      f =>
+        Aggregates.filter(
+          Filters.eq(fieldName = f._1, value = f._2)
+        )
+    )
+
+    val projection = Aggregates.project(
+      Projections.fields(
+        Projections.include("data"),
+        Projections.exclude("_id")
+      )
+    )
+
+    Source.fromPublisher(
+      collection
+        .aggregate[BsonValue](Seq(standardFilters, projection) ++ extraFilters)
+        .allowDiskUse(true)
+        .map(Codecs.fromBson[JsObject](_))
+    )
+  }
```

Comment:
Possible refactor here. Essentially the same method but one has an empty `extraFilters`